plasmapy/simulation/particle_tracker.py

```diff
@@ -169,6 +170,89 @@ def total(self) -> float:
         return float(self.tracker.nparticles)
 
 
+class ParticlesPushedThroughGridsTerminationCondition(AbstractTerminationCondition):
```

Comment:
"AllParticlesOffGridTerminationCondition"? Since all these are for particles being pushed through a grid?